src/Common/MemoryTracker.cpp

```diff
@@ -192,7 +193,7 @@ void MemoryTracker::allocImpl(Int64 size, bool throw_if_memory_exceeded, MemoryT
     }
 
     std::bernoulli_distribution fault(fault_probability);
-    if (unlikely(fault_probability > 0.0 && fault(thread_local_rng)))
+    if (unlikely(fault_probability > 0.0 && fault(thread_local_rng) || MemoryTrackerFaultInjectorInThread::faulty()))
```

Comment:
Just small optimization. I guess it should also be inside unlikely

Reply:
It is already inside unlikely.

But TBH, I would consider this an optimization for reader, in other words to make it more obvious what is the regular code path for reader, not for processor, since it is not that hot place.